common/go-utils/src/inv.rs

```diff
@@ -103,20 +87,18 @@ pub fn list_upstream_artifacts() -> Result<Vec<Artifact>, String> {
         .map_err(|e| e.to_string())?
         .iter()
         .flat_map(|release| &release.files)
-        .filter(|file| !file.sha256.is_empty())
-        .filter_map(|file| {
-            file.target_arch()
-                .filter(|target| target == ARCH)
-                .map(|target| {
-                    Version::parse_go(&file.version)
-                        .map(|version| Artifact {
-                            go_version: file.version.clone(),
-                            semantic_version: version,
-                            architecture: target,
-                            sha_checksum: file.sha256.clone(),
-                        })
-                        .map_err(|e| e.to_string())
+        .filter(|file| !file.sha256.is_empty() && file.os == "linux" && file.arch == "amd64")
+        .map(|file| {
+            Version::parse_go(&file.version)
```

Comment:
not required but this seems like a good candidate for a [TryInto](https://doc.rust-lang.org/std/convert/trait.TryInto.html) impl.  E.g.;

```rust
impl <'a> TryFrom<&'a GoFile> for Artifact {
    type Error = VersionParseError;

    fn try_from(value: &'a GoFile) -> Result<Self, Self::Error> {
        let version = Version::parse_go(&value.version)?;
        Ok(Artifact {
            go_version: value.version.clone(),
            semantic_version: version,
            os: value.os.clone(),
            arch: value.arch.clone(),
            sha_checksum: value.sha256.clone(),
            url: format!("{}/{}", GO_HOST_URL, value.filename),
        })
    }
}
```

and you'd use it here like this:

```rust
.map(|go_file| Artifact::try_from(go_file).map_err(|e| e.to_string()))
```

Reply:
Much appreciated! I added it here https://github.com/heroku/buildpacks-go/pull/229/commits/3c70b3c03c378c5cf54131e6bc9762fd7c91fb61